### PR TITLE
build: add `rsdoctor` for bundle analysis

### PR DESCRIPTION
**Pull request by anonrig**
Newly released `rsdoctor` package provides really good insights for bundle analysis. You can enable it by `RSDOCTOR=1 yarn build`.

Here are some screenshots:

Screenshot 2024-01-25 at 1 57 56 PM
Screenshot 2024-01-25 at 1 57 22 PM
Screenshot 2024-01-25 at 1 56 25 PM

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -10909,6 +11607,25 @@ webidl-conversions@^7.0.0:
   resolved "https://registry.yarnpkg.com/webidl-conversions/-/webidl-conversions-7.0.0.tgz#256b4e1882be7debbf01d05f0aa2039778ea080a"
   integrity sha512-VwddBukDzu71offAQR975unBIGqfKZpM+8ZX6ySk8nYhVoo5CYaZyzt3YBvYtRtO+aoGlqxPg/B87NGVZ/fu6g==
 
+webpack-bundle-analyzer@^4.9.1:
```
Comment (scttcper) [inline yarn.lock:11610]: interesting that its a sub dependency

File: yarn.lock [hunk 2]
```diff
@@ -11035,6 +11752,36 @@ webpack@5.87.0:
     watchpack "^2.4.0"
     webpack-sources "^3.2.3"
 
+webpack@^5, webpack@^5.89.0:
```
Comment (scttcper) [inline yarn.lock:11755]: is this a duplicate?

Comment (scttcper) [reply yarn.lock:11755]: looks like we'd have 2 versions of webpack

Comment (anonrig) [reply yarn.lock:11755]: Maybe it's time to move away from Yarn to either npm or pnpm.

File: yarn.lock (hunk 1 above)
Comment (ScriptedAlchemy) [reply yarn.lock:11610]: I think we render bundle analyzer visualization somewhere in rsdoctor, visual is good just not great for inspection

File: yarn.lock (hunk 2 above)
Comment (anonrig) [reply yarn.lock:11755]: > 'yarn dedupe' should solve this.

It didn't

Comment (scttcper) [reply yarn.lock:11755]: try `npx yarn-deduplicate && yarn install`

Comment (anonrig) [reply yarn.lock:11640]: Today I learned :-)

**Review by scttcper — APPROVED** [review]
really excited about the duplicate detection, seems interesting overall

**Pull request merged**